### PR TITLE
Wait for Nuget restores to be complete before considering the solution fully loaded.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This affects NavigateTo and Lightbulbs, both of which wait for 'fully loaded' before providing results.  For both of these we def want to wait for restores as prior to a restoration we can't really provide good results since we don't know the full meaning of hte code, and because it's likely that doing this work while the restore is happening will just cause contention.

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs [hunk 1]
```diff
@@ -113,11 +115,17 @@ public async Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
...
-                return !status.IsInProgress;
+                if (status.IsInProgress)
+                    return false;
+
+                var componentModel =
+                    await _serviceProvider.GetServiceAsync<SComponentModel, IComponentModel>(throwOnFailure: false)
+                                          .WithCancellation(cancellationToken).ConfigureAwait(false);
+
+                var solutionRestoreStatusProvider = componentModel.GetService<IVsSolutionRestoreStatusProvider>();
+                return await solutionRestoreStatusProvider.IsRestoreCompleteAsync(cancellationToken).ConfigureAwait(false);
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs:128]: 📝 We should verify with NuGet team that this is actually needed before merging it

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs:128]: @emgarten Can you advise here?  We're trying to find the point that we can consider a solution reasonably "fully loaded".  i.e. that the language service has been told about all the real files/dlls necessary for normal IDE features to work properly.  The above attemps to wait until we're past the intellisense-build stage and that the nuget-restore has completed.  Do you think that's sufficient/accurate?  Thanks!

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs:128]: @nkolev92 Was told you'd be good to pull in on this :)

Comment (nkolev92) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs:128]: I think the preference would be use to Operation Progress APIs. 
cc @davkean 

The operation progress captures the fact that NuGet restore has run.

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs:128]: so we are using that up on the line where we do:

```
                if (status.IsInProgress)
                    return false;
```

However, that seems to be about knowing the state of hte intellisense build.  This is more about knowing when nuget restore has completed.

**Comment by CyrusNajmabadi** [comment]
Closing out, this is unnecessary, our existing check is sufficient.

**Comment by davkean** [comment]
To add to what we summarized internally, operation progress is that indicator:

```
            var operationProgressStatus = await GetGlobalServiceAsync<SVsOperationProgress, IVsOperationProgressStatusService>();
            var stageStatus = operationProgressStatus.GetStageStatus(CommonOperationProgressStageIds.Intellisense);
            await stageStatus.WaitForCompletionAsync();
```

This waits for all in progress design-time builds and NuGet restore to complete and for those changes to be pushed to Roslyn.

**Pull request closed without merging**